src/runtime_src/tools/xclbinutil/unittests/AieTraceMetadata/SectionAieTraceMetadata.py

```diff
@@ -0,0 +1,82 @@
+from argparse import RawDescriptionHelpFormatter
+import argparse
+import os
+import subprocess
+
+# Start of our unit test
+# -- main() -------------------------------------------------------------------
+#
+# The entry point to this script.
+#
+# Note: It is called at the end of this script so that the other functions
+#       and classes have been defined and the syntax validated
+def main():
+  # -- Configure the argument parser
+  parser = argparse.ArgumentParser(formatter_class=RawDescriptionHelpFormatter, description='description:\n  Unit test wrapper for the AIE TRACE METADATA section')
+  parser.add_argument('--resource-dir', nargs='?', default=".", help='directory containing data to be used by this unit test')
+  args = parser.parse_args()
+
+  # Validate that the resource directory is valid
+  if not os.path.exists(args.resource_dir):
+      raise Exception("Error: The resource-dir '" + args.resource_dir +"' does not exist")
+
+  if not os.path.isdir(args.resource_dir):
+      raise Exception("Error: The resource-dir '" + args.resource_dir +"' is not a directory")
+
+  # Prepare for testing
+  xclbinutil = "xclbinutil"
+
+  # Start the tests
+  print ("Starting test")
+
+  # ---------------------------------------------------------------------------
+
+  step = "1) Read in the AIE TRACE METADATA section"
+
+  inputJSON = os.path.join(args.resource_dir, "aie_trace_config.json")
+
+  cmd = [xclbinutil, "--add-section", "AIE_TRACE_METADATA:JSON:" + inputJSON]
+  execCmd(step, cmd)
+
+  # ---------------------------------------------------------------------------
```

Comment:
After reading it in, it should be written out and correctness should be verified.

See the unit test for PSKernels as a reference: https://github.com/Xilinx/XRT/blob/master/src/runtime_src/tools/xclbinutil/unittests/PSKernel/PSKernel.py